caravel/assets/javascripts/SqlLab/components/QuerySearchBar.jsx

```diff
@@ -0,0 +1,147 @@
+const $ = window.$ = require('jquery');
+import React from 'react';
+
+import { Button } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Select from 'react-select';
+import * as Actions from '../actions';
+
+class QuerySearchBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      databaseLoading: false,
+      databaseOptions: [],
+      userLoading: false,
+      userOptions: [],
+      databaseId: null,
+      userId: null,
+      searchText: null,
+      sql: null,
+      queryState: 1,
+    };
+  }
+  componentWillMount() {
+    this.fetchDatabaseOptions();
+    this.fetchUsers();
+  }
+  changeDb(db) {
+    const val = (db) ? db.value : null;
+    this.setState({ databaseId: val });
+  }
+  changeUser(user) {
+    const val = (user) ? user.value : null;
+    this.setState({ userId: val });
+  }
+  changeStatus(status) {
+    const val = (status) ? status.value : null;
+    this.setState({ queryState: val });
+  }
+  fetchDatabaseOptions() {
```

Comment:
Once we get DatabaseSelect as a child component, the select value would have to be passed from it back to QuerySearch or QueryEditor. Do we want to pass the value from child to parent?